internal/provider/cluster_resource.go

```diff
@@ -439,6 +438,23 @@ func (r *clusterResource) Create(
 			}
 		}
 
+		if IsKnown(plan.CockroachVersion) {
+			var planType client.PlanType
+			if clusterSpec.Plan != nil {
```

Comment:
This plan type derivation logic seems like something that should be a utility function.

Reply:
Okay, I've moved it out to its own function and added a test for it.